app/controllers/sources_controller.rb

```diff
@@ -0,0 +1,61 @@
+class SourcesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_source, only: [:show, :edit, :update]
+
+  # GET /ext_apps
+  def index
+    @sources = Source.all
+  end
+
+  # GET /remotes/1
+  def show
+  end
+
+  # GET /sources/new
+  def new
+    @source = Source.new
+  end
+
+  # GET /remotes/1/edit
+  def edit
+  end
+
+  # PATCH/PUT /ext_apps/1
+  def update
+    @source = Source.find(params[:id])
+
+    if @source.update(source_params)
+      redirect_to @source
+    else
+      render 'edit'
+    end
+  end
+
+  # POST /sources
+  def create
+    @source = Source.create(source_params)
+
+    respond_to do |format|
+      if @source.errors.none?
```

Comment:
why not use similar pattern with other controllers?
`#new`
then `if @remote.save` below

Reply:
Noted

Reply:
Done. https://github.com/pathfinder-cm/pathfinder-mono/pull/26/commits/a7e9687e061cf8a7b413431583b6af262206507a